app/models/order_cycle.rb

```diff
@@ -8,6 +8,14 @@ class OrderCycle < ActiveRecord::Base
 
   has_many :exchanges, :dependent => :destroy
 
+  # These scope names are prepended with "cached_" because there are existing accessor methods
+  # :incoming_exchanges and :outgoing_exchanges. Future work can be done to name these properly.
+  has_many :cached_incoming_exchanges, conditions: { incoming: true }, class_name: "Exchange"
+  has_many :cached_outgoing_exchanges, conditions: { incoming: false }, class_name: "Exchange"
+
+  has_many :suppliers, source: :sender, through: :cached_incoming_exchanges, uniq: true
+  has_many :distributors, source: :receiver, through: :cached_outgoing_exchanges, uniq: true
```

Comment:
can you explain a little bit what's the difference between the SQL resulting of this and the previous implementation?

Reply:
@sauloperez One effect is that `order_cycle.suppliers` and `order_cycle.distributors` would be cached, so running a method multiple times would query only once. This also makes it possible to use eager-loading, like: `order_cycle_scope.includes(:distributors, :suppliers)`

Reply:
looks good but I think we need to clean up a little bit.
the code and comments should be about the present. future work is represented on github.
can we move the references to the future to github tech debt issue?

there are two:
1 - "Future work can be done to name these properly."
2 - "TODO: DRY the incoming/outgoing clause used in several cases below" (this TODO is 5 years old, which is a common thing for TODOs)

even if we dont create the tech debt issue I think both points should be removed from the code.

Reply:
Regarding this old comment:

```
# TODO: DRY the incoming/outgoing clause used in several cases below
# See Spree::Product definition, scopes variants and variants_including_master
# This will require these accessors to be renamed
```

I checked these and didn't get useful information to understand the accessors, so I removed the entire comment.

Changed here: https://github.com/openfoodfoundation/openfoodnetwork/pull/3802/commits/c0a5bcce92b4a22ce19b1e5cf3ea0391ffa2c79a